test/message/console_low_stack_space.js

```diff
@@ -6,17 +6,26 @@ global.console = {};
 
 require('../common');
 
+// This test checks that, if Node cannot put together the `console` object
+// because it is low on stack space while doing so, it can succeed later
+// once the stack has unwound a little, and `console` is in a usable state then.
+
+let compiledConsole;
+
 function a() {
   try {
     return a();
   } catch (e) {
-    const console = consoleDescriptor.get();
-    if (console.log) {
-      console.log('Hello, World!');
+    compiledConsole = consoleDescriptor.get();
```

Comment:
Could you add a check that `e.message === 'Maximum call stack size exceeded'` (should be `assert`, but we can assume the stack is full)